src/lib/crypto/DY.Lib.Crypto.AEAD.Split.fst

```diff
@@ -0,0 +1,79 @@
+module DY.Lib.Crypto.AEAD.Split
+
+open Comparse // for_allP, for_allP_eq
+open DY.Core
+open DY.Lib.SplitFunction
+
+let split_aead_predicate_params (cusages:crypto_usages) = {
+  singleton_split_function_parameters string with
+
+  tagged_data_t = (trace & (key:bytes{AeadKey? (get_usage key)}) & bytes & bytes);
+  raw_data_t = (trace & (key:bytes{AeadKey? (get_usage key)}) & bytes & bytes);
+  output_t = prop;
+
+  decode_tagged_data = (fun (tr, key, msg, ad) ->
+    let AeadKey tag _ = get_usage key in
+    Some (tag, (tr, key, msg, ad))
+  );
```

Comment:
(applies for all of these functions) It seems like we might in principle want this to be more general in some cases, perhaps also looking at the `data` field or the data being encrypted to decode the "tag". Probably what's written here is the typical use case, so I think this is good for now, but we may want to also consider what it looks like to modify the choice of tag/decoding function. Of course one can always define a protocol-specific instance of this type, but maybe there are alternatives?

On further reading, it seems to me like a protocol-specific instance if needed is the correct approach.

Reply:
I think this should be done in a protocol-specific instance indeed!